inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/convolution/convolution_kernel_b_fs_yx_fsv_16_32_imad_dw.cpp

```diff
@@ -123,21 +111,137 @@ ConvolutionKernel_b_fs_yx_fsv_16_32_imad_dw::GetAutoTuneParams(const convolution
+    bool selected = false;
+    // Helper function, selecting specified auto tune params if they are correct for current hw
+    auto try_to_select = [&](size_t simd, size_t tile_x, size_t lws0, size_t lws1, bool preload_input_slm, std::string exe_mode) -> bool {
+        tune_params.simd = simd;
+        tune_params.tile_x = tile_x;
+        tune_params.lws0 = lws0;
+        tune_params.lws1 = lws1;
+        tune_params.preload_input_slm = preload_input_slm;
+        tune_params.exeMode = exe_mode;
+        selected = ValidateAutoTuneParams(params, tune_params);
+        return selected;
+    };
+    // List of fine-tuned known best cases
+    bool filter_3x3 = params.filterSize.x == 3 && params.filterSize.y == 3;
+    bool dilation_1x1 = params.dilation.x == 1 && params.dilation.y == 1;
+    bool stride_1x1 = params.stride.x == 1 && params.stride.y == 1;
+    bool stride_2x2 = params.stride.x == 2 && params.stride.y == 2;
+    // Filter 3x3 with stride 1x1
+    if (fsv == 16 && filter_3x3 && stride_1x1 && dilation_1x1 && output.X().v == 75 && output.Y().v == 75)
+        try_to_select(16, 15, 1, 4, true, DEFAULT);
+
+    // Filter 3x3 with stride 2x2
+    if (fsv == 16 && filter_3x3 && stride_2x2 && dilation_1x1 && output.X().v == 75 && output.Y().v == 75)
+        try_to_select(16, 15, 1, 16, true, DEFAULT);
+
+    // Check if SLM can provide data reuse for current parameters
+    bool use_slm_x = (params.filterSize.x - 1) * params.dilation.x + 1 >= params.stride.x;
+    bool use_slm_y = (params.filterSize.y - 1) * params.dilation.y + 1 >= params.stride.y;
+    // Small spatials are inefficent with SLM due to additional overhead
+    bool use_slm_size = output.X().v >= default_tile_x && output.Y().v >= default_tile_x;
+
+    if (!selected && use_slm_y && use_slm_x && use_slm_size) {
+        size_t tile_x = default_tile_x / 2;
+        size_t lws1 = 4;
+        size_t lws0 = 1;
+        // First try to select optimal group size in y dimension as it provides most data reuse
+        for (size_t c_lws1 = 4; c_lws1 <= 8; ++c_lws1) {
+            if (Pad(output.Y().v, c_lws1) < Pad(output.Y().v, lws1))
+                lws1 = c_lws1;
+            else if (Pad(output.Y().v, c_lws1) == Pad(output.Y().v, lws1) && c_lws1 % 2 == 0)
+                lws1 = c_lws1;
+        }
+        // For best hw utilization work-group size should be multiple of 2, so if y isn't force it in x
+        if (lws1 % 2 != 0) {
+            lws0 = 2;
+        }
+        // Local memory allocation works at 1kb granularity
+        auto calc_slm_size_kb = [&](size_t tile_x, size_t lws0, size_t lws1) {
+            size_t slm_x = (tile_x * lws0 - 1) * params.stride.x + (params.filterSize.x - 1) * params.dilation.x;
+            size_t slm_y = (lws1 == 1) ? params.filterSize.x
+                                       : (lws1 - 1) * params.stride.y + (params.filterSize.y - 1) * params.dilation.y + 1;
+            return CeilDiv(slm_x * slm_y * fsv, 1024);
```

Comment:
As I know minimum SLM allocation size is 4k per workgroup. I amn't sure but maybe it makes sence to check this limitation in your code?

return slm_x * slm_y * fsv < 4096 ? 4096 : CeilDiv(slm_x * slm_y * fsv, 1024);

Reply:
That's not exactly true, there is no problem with allocating 1kb of slm.
This is just easy explanation why we can't have 64 work-groups using 1kb slm each - we can have only up to 16 work-groups executing concurrently.
I actually approximate more detailed tradeoff between slm usage and number of threads in 
```
            bool c_slm_size = calc_slm_size_kb(c_tile, lws0, lws1) <= lws0 * lws1;
```
and by ensuring at least 4 sub-groups per work-group are used.

Reply:
Thank you for a good explanation.